rfd/0099-link-bundled-tsh.md

```diff
@@ -0,0 +1,127 @@
+file](https://rpm-packaging-guide.github.io/#what-is-a-spec-file) for its rpm package. As shown by
+[microsoft/vscode#142907](https://github.com/microsoft/vscode/pull/142907), this lets rpm correctly
+recognize that the `code` binary is owned by the VSCode package. This also means that rpm will
+automatically clean up this file when the user uninstalls the package.
+
+The deb package accomplishes the same goal through post install and post removal scripts.
+
+* https://github.com/microsoft/vscode/blob/1.74.0/resources/linux/debian/postinst.template#L6-L8
+* https://github.com/microsoft/vscode/blob/1.74.0/resources/linux/debian/postrm.template#L6
+* https://github.com/microsoft/vscode/blob/1.74.0/resources/linux/rpm/code.spec.template#L34
+
+#### Windows
+
+During installation, the directory with `code` is added to the `Path` env variable.
+
+* https://github.com/microsoft/vscode/blob/1.74.0/build/win32/code.iss#L1299
+
+#### macOS
+
+The command palette offers [separate
+commands](https://code.visualstudio.com/docs/setup/mac#_launching-from-the-command-line) for
+installing and uninstalling the CLI tool. The installation command executes an AppleScript with
+administrator privileges which symlinks `code` to `/usr/local/bin/code`.
+
+* https://github.com/microsoft/vscode/blob/1.74.0/src/vs/workbench/electron-sandbox/actions/installActions.ts
+* https://github.com/microsoft/vscode/blob/1.74.0/src/vs/platform/native/electron-main/nativeHostMainService.ts#L261
+
+### How electron-builder does it
+
+#### Linux
+
+Linux is the only platform where during the installation of a package, electron-builder
+automatically symlinks the binary to `/usr/bin`.
+
+Unlike VSCode, electron-builder doesn't have special templates for any particular package format.
+Instead it uses [fpm](https://fpm.readthedocs.io) which handles many different formats.
+
+The binary is linked through a post install and post removal scripts. This means that rpm doesn't
+recognize that the binary comes from the specific package.
+
+* https://github.com/electron-userland/electron-builder/blob/v24.0.0-alpha.5/packages/app-builder-lib/templates/linux/after-install.tpl
+* https://github.com/electron-userland/electron-builder/blob/v24.0.0-alpha.5/packages/app-builder-lib/templates/linux/after-remove.tpl
+
+### How Connect can do it
+
+#### Linux
+
+We can automatically link tsh using the same method that electron-builder uses. Unfortunately,
+electron-builder doesn't seem to provide a way to extend the post install script in any way. This
+most likely means that we'll have to copy the default script and keep it in sync when updating
+electron-builder.
+
+Instead of symlinking the binary to `/usr/bin/tsh`, we can symlink it to `/usr/local/bin/tsh`. This
+is where our Linux teleport package places our binaries. This way if someone installs Connect after
+installing teleport, we'll overwrite just a single symlink instead of the user ending up with two
```

Comment:
> This way if someone installs Connect after installing teleport

What happens if someone updates teleport after they've installed Connect? Could this mess with the protobuf files if the updated tsh contains changes that the Connect binary hasn't shipped with? We haven't had to worry about this with Connect yet since we always ship it with it's "own" tsh, but will this change cause a way to get out of sync?